aperturedb/cli/ingest.py

```diff
@@ -214,3 +218,62 @@ def from_csv(filepath: Annotated[str, typer.Argument(
     else:
         loader.ingest(data, batchsize=batchsize,
                       numthreads=num_workers, stats=stats)
+
+
+@app.command()
+def generate_embedding_csv_from_image_csv(
+    input_file: Annotated[str, typer.Argument(
+        help="Path to csv for ingestion")],
+    set_name: Annotated[str, typer.Argument(
+        help="Descriptor set name to be associated with the embeddings")],
+    transformer: Annotated[Optional[List[TransformerType]], typer.Option(
+        help="Apply transformer to the pipeline [Can be specified multiple times]")] = None,
+    user_transformer: Annotated[Optional[List[str]], typer.Option(
+        help="Apply user transformer to the pipeline as path to file [Can be specified multiple times.]")] = None,
+    sample_count: Annotated[int, typer.Option(
+        help="Number of samples to ingest (-1 for all)")] = -1,
+):
+    """
+    Create a DescriptorDataCSV from an ImageDataCSV, using the specified transformer.
+    Also save embeddings in a separate file, and generate the ConnectionDataCSV
+    for linking the images to the descriptors.
+    """
+    data = ImageDataCSV(input_file)
+    data.sample_count = len(data) if sample_count == -1 else sample_count
+    if transformer or user_transformer:
+        transformer = transformer or []
+        user_transformer = user_transformer or []
+        all_transformers = transformer + user_transformer
+        if len(all_transformers) == 0:
+            console.log(
+                "No transformer specified . Generating embeddings from raw images requires a transformer.")
+            typer.Abort()
+        data = _apply_pipeline(data, all_transformers,
+                               adb_data_source=f"{os.path.basename(input_file)}")
+    filename = f"{os.path.basename(input_file)}_{all_transformers[-1]}"
+    metadata = []
+    connection = []
+    embeddings = []
+
+    for i in tqdm(range(data.sample_count)):
+        d = data[i]
+        embeddings.append(np.frombuffer(d[1][1], dtype=np.float32))
+        metadata.append({
```

Comment:
If speed becomes (or is) an issue, generating [pandas df columns directly](https://github.com/aperture-data/demos/blob/master/coco/generate_coco_csv.py#L26) may be faster.

Reply:
There is speed thing, but it's because of model's inference. For now, this is mostly a util to avoid run time inference cost.